File: sudoku-game/game-boot/src/main/java/com/zumin/sudoku/game/mapper/GameRecordMapper.java
package com.zumin.sudoku.game.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.zumin.sudoku.game.pojo.entity.GameRecord;
import java.time.LocalDate;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

@Mapper
public interface GameRecordMapper extends BaseMapper<GameRecord> {

  Integer countByDateBetween(@Param("startDate") LocalDate startDate, @Param("endDate") LocalDate endDate);
}

File: sudoku-game/game-boot/src/main/java/com/zumin/sudoku/game/mapper/GameRecordMapper.kt
package com.zumin.sudoku.game.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import com.zumin.sudoku.game.pojo.GameRecord
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import java.time.LocalDate

@Mapper
interface GameRecordMapper : BaseMapper<GameRecord> {
  fun countByDateBetween(@Param("startDate") startDate: LocalDate, @Param("endDate") endDate: LocalDate): Int
}